### PR TITLE
Pushing monitoring details only for managed nodes

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
tendrl-bug-id: Tendrl/node-agent#810
bugzilla: 1564510

Signed-off-by: GowthamShanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
@nthomas-redhat @r0h4n please review

File: tendrl/node_agent/monitoring/collectd/collectors/gluster/heavy_weight/tendrl_gluster_heal_info.py [hunk 1]
```diff
@@ -23,7 +23,12 @@ def _parse_heal_info_stats(tree, integration_id, etcd_client):
         brick_host = tendrl_glusterfs_utils.find_brick_host(
             etcd_client, integration_id, brick_host
         )
-
+        # if brick node is not managed then don't
+        # consider the brick
+        if not tendrl_glusterfs_utils.find_node_managed(
+            etcd_client, brick_host, integration_id
+        ) == "yes":
```
Comment (shtripat) [inline tendrl/node_agent/monitoring/collectd/collectors/gluster/heavy_weight/tendrl_gluster_heal_info.py]: Use `!=` I feel

File: tendrl/node_agent/monitoring/collectd/collectors/gluster/utils.py [hunk 2]
```diff
@@ -352,3 +352,21 @@ def find_brick_host(etcd_client, integration_id, brick_host):
...
+                key = "%s/NodeContext/data" % node_id.key
+                data = etcd_client.read(key).value
+                data = json.loads(data)
+                if brick_host == data.get("fqdn", ""):
+                    return data.get("is_managed", None)
+        except (urllib3.exceptions.TimeoutError, etcd.EtcdKeyNotFound):
+            _msg = "Error finding managed status for brick" \
+                   "node %s" % brick_host
+            collectd.warning(_msg)
+            collectd.warning(traceback.format_exc())
```
Comment (shtripat) [inline tendrl/node_agent/monitoring/collectd/collectors/gluster/utils.py]: Shouldn't we return some value here?

File: tendrl/node_agent/monitoring/collectd/collectors/gluster/heavy_weight/tendrl_gluster_heal_info.py (hunk 1 above)
Comment (GowthamShanmugam) [reply tendrl/node_agent/monitoring/collectd/collectors/gluster/heavy_weight/tendrl_gluster_heal_info.py]: ok

File: tendrl/node_agent/monitoring/collectd/collectors/gluster/utils.py (hunk 2 above)
Comment (GowthamShanmugam) [reply tendrl/node_agent/monitoring/collectd/collectors/gluster/utils.py]: no need, if nothing is returned then receivers default value is None

File: tendrl/node_agent/monitoring/collectd/collectors/gluster/utils.py [hunk 3]
```diff
@@ -352,3 +352,22 @@ def find_brick_host(etcd_client, integration_id, brick_host):
             _msg = "Error finding fqdn/ip for brick %s" % brick_host
             collectd.warning(_msg)
             collectd.warning(traceback.format_exc())
+
+
+def find_node_managed(etcd_client, brick_host, integration_id):
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply tendrl/node_agent/monitoring/collectd/collectors/gluster/utils.py]: ok i will change

**Comment by GowthamShanmugam** [comment]
@r0h4n @shtripat @nthomas-redhat This patch is replaced by /Tendrl/node-agent/pull/815, so this patch is not required, i am closing this

**Comment by GowthamShanmugam** [comment]
@r0h4n @shtripat @nthomas-redhat please review

**Pull request closed without merging**